File: lab2/outputFiles/1.py

def displayMenu():
	print("=== CALCULATOR ===" )
	print("1. Addition (+)" )
	print("2. Subtraction (-)" )
	print("3. Multiplication (*)" )
	print("4. Division (/)" )
	print("5. Power (^)" )
	print("6. Exit" )
	print("Choose operation: ", end='')

def calculator():
	choice = 0
	num1 = 0
	num2 = 0
	result = 0
	
	while choice != 6:
		displayMenu()
		choice = float(input())
		
		match choice:
			case 1:
				print("Enter first number: ", end='')
				num1 = float(input())
				print("Enter second number: ", end='')
				num2 = float(input())
				result = num1 + num2
				print("Result: " , num1 , " + " , num2 , " = " , result )
			
			case 2:
				print("Enter first number: ", end='')
				num1 = float(input())
				print("Enter second number: ", end='')
				num2 = float(input())
				result = num1 - num2
				print("Result: " , num1 , " - " , num2 , " = " , result )
			
			case 3:
				print("Enter first number: ", end='')
				num1 = float(input())
				print("Enter second number: ", end='')
				num2 = float(input())
				result = num1 * num2
				print("Result: " , num1 , " * " , num2 , " = " , result )
			
			case 4:
				print("Enter dividend: ", end='')
				num1 = float(input())
				print("Enter divisor: ", end='')
				num2 = float(input())
				if num2 != 0:
					result = num1 / num2
					print("Result: " , num1 , " / " , num2 , " = " , result )
				else:
					print("Error! Division by zero is impossible." )
			
			case 5:
				print("Enter base: ", end='')
				num1 = float(input())
				print("Enter exponent: ", end='')
				num2 = float(input())
				result = pow(num1, num2)
				print("Result: " , num1 , " ^ " , num2 , " = " , result )
			case 6:
				print("Goodbye!" )
			case _:
				print("Invalid choice! Please try again." )
		
		print()

if __name__ == '__main__':
	print("Welcome to the calculator!" )
	calculator()
	

File: lab2/testFiles/1.cpp
#include <iostream>
#include <cmath>

void displayMenu() {
    std::cout << "=== CALCULATOR ===" << std::endl;
    std::cout << "1. Addition (+)" << std::endl;
    std::cout << "2. Subtraction (-)" << std::endl;
    std::cout << "3. Multiplication (*)" << std::endl;
    std::cout << "4. Division (/)" << std::endl;
    std::cout << "5. Power (^)" << std::endl;
    std::cout << "6. Exit" << std::endl;
    std::cout << "Choose operation: ";
}

void calculator() {
    int choice;
    float num1, num2, result;
    
    while (choice != 6){
        displayMenu();
        std::cin >> choice;
        
        switch (choice) {
            case 1:
                std::cout << "Enter first number: ";
                std::cin >> num1;
                std::cout << "Enter second number: ";
                std::cin >> num2;
                result = num1 + num2;
                std::cout << "Result: " << num1 << " + " << num2 << " = " << result << std::endl;
                break;
                
            case 2:
                std::cout << "Enter first number: ";
                std::cin >> num1;
                std::cout << "Enter second number: ";
                std::cin >> num2;
                result = num1 - num2;
                std::cout << "Result: " << num1 << " - " << num2 << " = " << result << std::endl;
                break;
                
            case 3:
                std::cout << "Enter first number: ";
                std::cin >> num1;
                std::cout << "Enter second number: ";
                std::cin >> num2;
                result = num1 * num2;
                std::cout << "Result: " << num1 << " * " << num2 << " = " << result << std::endl;
                break;
                
            case 4:
                std::cout << "Enter dividend: ";
                std::cin >> num1;
                std::cout << "Enter divisor: ";
                std::cin >> num2;
                if (num2 != 0) {
                    result = num1 / num2;
                    std::cout << "Result: " << num1 << " / " << num2 << " = " << result << std::endl;
                } 
                else {
                    std::cout << "Error! Division by zero is impossible." << std::endl;
                }
                break;
                
            case 5:
                std::cout << "Enter base: ";
                std::cin >> num1;
                std::cout << "Enter exponent: ";
                std::cin >> num2;
                result = std::pow(num1, num2);
                std::cout << "Result: " << num1 << " ^ " << num2 << " = " << result << std::endl;
                break;
            case 6:
                std::cout << "Goodbye!" << std::endl;
                break;
            default:
                std::cout << "Invalid choice! Please try again." << std::endl;
                break;
        }
        
        std::cout << std::endl;
    }
}

int main() {
    std::cout << "Welcome to the calculator!" << std::endl;
    calculator();
    
    return 0;
}